test/arranger-conduit/invariants/InvariantTestBase.t.sol

```diff
@@ -0,0 +1,189 @@
+            uint256 sumWithdrawableFunds;
+            uint256 sumWithdrawals;
+
+            for (uint256 j = 0; j < ilks.length; j++) {
+                sumDeposits          += conduit.deposits(ilks[j], assets[i]);
+                sumRequestedFunds    += conduit.requestedFunds(ilks[j], assets[i]);
+                sumWithdrawableFunds += conduit.withdrawableFunds(ilks[j], assets[i]);
+                sumWithdrawals       += conduit.withdrawals(ilks[j], assets[i]);
+            }
+
+            assertEq(conduit.totalDeposits(assets[i]),          sumDeposits);
+            assertEq(conduit.totalRequestedFunds(assets[i]),    sumRequestedFunds);
+            assertEq(conduit.totalWithdrawableFunds(assets[i]), sumWithdrawableFunds);
+            assertEq(conduit.totalWithdrawals(assets[i]),       sumWithdrawals);
+        }
+    }
+
+    function assert_invariant_E() internal {
+        for (uint256 i = 0; i < assets.length; i++) {
+            MockERC20 asset = MockERC20(assets[i]);
+
+            assertGe(asset.balanceOf(address(conduit)), conduit.totalWithdrawableFunds(assets[i]));
+        }
+    }
+
+    function assert_invariant_F() internal {
+        IArrangerHandlerLike arrangerHandler_ = IArrangerHandlerLike(arrangerHandler);
+        for (uint256 i = 0; i < assets.length; i++) {
+            assertEq(
+                conduit.totalWithdrawableFunds(assets[i]),
+                arrangerHandler_.returnedFunds(assets[i]) - conduit.totalWithdrawals(assets[i])
+            );
+        }
+    }
+
+    // NOTE: Interesting finding, if there are transfers and returnFunds calls before
+    //       the first deposit, there can be a situation where returnedFunds > totalDeposits
+    //       very early in a sequence.
+    // NOTE: Had to add a transferredFunds ghost variable because drawnFunds can actually be higher
+    //       than totalDeposits and returnedFunds if transfer + draw happens early enough.
+    function assert_invariant_G() internal {
+        IArrangerHandlerLike   arrangerHandler_   = IArrangerHandlerLike(arrangerHandler);
+        ITransfererHandlerLike transfererHandler_ = ITransfererHandlerLike(transfererHandler);
+
+        for (uint256 i = 0; i < assets.length; i++) {
+            uint256 netBalance =
+                conduit.totalDeposits(assets[i])
+                + arrangerHandler_.returnedFunds(assets[i])
+                + transfererHandler_.transferredFunds(assets[i])
+                - arrangerHandler_.drawnFunds(assets[i])
+                - conduit.totalWithdrawals(assets[i]);
+
+            assertEq(MockERC20(assets[i]).balanceOf(address(conduit)), netBalance);
+        }
+    }
+
+    function assert_invariant_H() internal {
+        for (uint256 i = 0; i < assets.length; i++) {
+            assertEq(
+                MockERC20(assets[i]).balanceOf(address(conduit)),
+                conduit.availableFunds(assets[i]) + conduit.totalWithdrawableFunds(assets[i])
+            );
+        }
+    }
```

Comment:
Since `availableFunds` returns balance - withdrawable funds, is this invariant really adding anything of value? If you want to leave it in that's fine as it doesn't really detract either.

Reply:
Yeah I was thinking this too, but I like keeping this kind of semi redundant stuff in in these tests